app/services/user_treatment.rb

```diff
@@ -0,0 +1,19 @@
+class UserTreatment
+
+  def call(user: user, type: type)
+    @type = type
+    relation = Hash.new {|h,k| h[k] = @type }
+    relation[@type.class.to_s.underscore]
+    @user  = user
+    @trim  = user.treatments.build(relation)
+
+    trim_facial_hair if type.is_a? TrimType
+    @trim
+  end
```

Comment:
(1) I have to jump through a lot of hoops to figure out what this method does.  It is vastly over-generalized.  Don't do that.  It makes code that is really difficult to figure out what it does in concrete situations which is actually what we care about.
(2) It appears you wrote this generalization to later apply something other than a trim to your users... being a massage.  Maybe I missed it, but it appears you did not get around to writing the portion where a massage is applied via this method.  Also not a good idea.  Don't try and DRY or generalize things until you have 2, or ideally 3 useages of it.  Most of the time your pattern recognition up front will be wrong and you'll end up rewriting this thing the next time you use it anyhow.